tracer/src/Datadog.Trace/Debugger/SpanCodeOrigin/EndpointDetector.cs

```diff
@@ -0,0 +1,282 @@
+                return true;
+            }
+
+            baseTypeHandle = nextBaseTypeHandle;
+        }
+
+        return false;
+    }
+
+    private static bool HasAttributeFromSet(CustomAttributeHandleCollection attributes, MetadataReader reader, HashSet<string> attributeNames)
+    {
+        foreach (var attributeHandle in attributes)
+        {
+            string fullName;
+            var attribute = reader.GetCustomAttribute(attributeHandle);
+            if (attribute.Constructor.Kind == HandleKind.MemberReference)
+            {
+                var ctor = reader.GetMemberReference((MemberReferenceHandle)attribute.Constructor);
+                var attributeType = reader.GetTypeReference((TypeReferenceHandle)ctor.Parent);
+                fullName = GetFullTypeName(attributeType.Namespace, attributeType.Name, reader);
+            }
+            else if (attribute.Constructor.Kind == HandleKind.MethodDefinition)
+            {
+                var ctor = reader.GetMethodDefinition((MethodDefinitionHandle)attribute.Constructor);
+                var attributeType = reader.GetTypeDefinition(ctor.GetDeclaringType());
+                fullName = GetFullTypeName(attributeType.Namespace, attributeType.Name, reader);
+            }
+            else
+            {
+                continue;
+            }
+
+            if (attributeNames.Contains(fullName))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private static bool IsInheritFromTypes(TypeDefinition typeDef, MetadataReader reader, HashSet<string> baseTypeNames)
+    {
+        var baseTypeHandle = typeDef.BaseType;
+        while (!baseTypeHandle.IsNil)
+        {
+            if (TryGetBaseTypeInfo(baseTypeHandle, reader, out var baseTypeName, out var nextBaseTypeHandle))
+            {
+                if (baseTypeNames.Contains(baseTypeName!))
+                {
+                    return true;
+                }
+            }
+
+            if (baseTypeHandle.Kind != HandleKind.TypeDefinition)
+            {
+                break;
+            }
+
+            baseTypeHandle = nextBaseTypeHandle;
+        }
+
+        return false;
+    }
+
+    private static bool IsPageModelHandler(MethodDefinition methodDef, MetadataReader reader)
+    {
+        var name = reader.GetString(methodDef.Name);
+        return name.StartsWith("On", StringComparison.Ordinal) &&
+               (name.Equals("OnGet", StringComparison.Ordinal) ||
+                name.Equals("OnGetAsync", StringComparison.Ordinal) ||
+                name.Equals("OnPost", StringComparison.Ordinal) ||
+                name.Equals("OnPostAsync", StringComparison.Ordinal) ||
+                name.Equals("OnPut", StringComparison.Ordinal) ||
+                name.Equals("OnPutAsync", StringComparison.Ordinal) ||
+                name.Equals("OnDelete", StringComparison.Ordinal) ||
+                name.Equals("OnDeleteAsync", StringComparison.Ordinal));
```

Comment:
All the HttpVerbs are supported, this only checks a subset. Also be aware these methods could have the `[NonHandler]` attribute, in which case they're _not_ a handler 😅

Reply:
Fixed in 9ed9b594f6cef3f4d0d243beb703c3e18b4efc7e